receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -90,7 +90,6 @@ func (r *pReceiver) Start(_ context.Context, host component.Host) error {
 		r.cfg.PrometheusConfig.GlobalConfig.ExternalLabels,
 	)
 	r.scrapeManager = scrape.NewManager(&scrape.Options{}, logger, r.ocaStore)
-	r.ocaStore.SetScrapeManager(r.scrapeManager)
```

Comment:
I think ScrapeManager capability has been removed from prometheusreceiver. Basing on the comments in other section, it's needed for "OCAStore". I'm not sure if the latter is still relevant but wanted to make sure we're not removing some part of functionality

Reply:
Strike that, reverse it.  This PR removes the need for the scrape manager in the OCAStore, but the receiver still needs it as it is what handles scraping targets emitted by the discovery manager.

Reply:
I'm not sure I follow but as Anthony said, its not required for `OCAStore` anymore. But it still exists in `r.scrapeManager`. I am resolving this comment as I think it has been addressed, but please let me know if I am missing something.